src/main/java/seedu/address/model/order/Order.java

```diff
@@ -0,0 +1,106 @@
+package seedu.address.model.order;
+
+import static seedu.address.commons.util.CollectionUtil.requireAllNonNull;
+
+import java.util.Objects;
+
+import seedu.address.model.person.Address;
+import seedu.address.model.person.Name;
+import seedu.address.model.person.Phone;
+
+/**
+ * Represents an Order in the ReadyBakey.
+ * Guarantees: details are present and not null, field values are validated, immutable.
+ */
+public class Order {
+
+    // Identity fields
+    private final Name name;
+    private final Phone phone;
+
+    // Data fields
+    private final Address address;
+    private final Details details;
+
+    /**
+     * Every field must be present and not null.
+     */
+    public Order(Name name, Phone phone, Address address, Details details) {
+        requireAllNonNull(name, phone, address, details);
+        this.name = name;
+        this.phone = phone;
+        this.address = address;
+        this.details = details;
+    }
+
+    public Name getName() {
+        return name;
+    }
+
+    public Phone getPhone() {
+        return phone;
+    }
+
+    public Details getDetails() {
+        return details;
+    }
+
+    public Address getAddress() {
+        return address;
+    }
+
+    /**
+     * Returns true if both orders have the same phone number.
+     * This defines a weaker notion of equality between two orders.
+     */
+    public boolean isSameOrder(Order otherOrder) {
+        if (otherOrder == this) {
+            return true;
+        }
+
+        return otherOrder != null
+                && otherOrder.getPhone().equals(getPhone());
```

Comment:
Great catch! I agree with @darshan152 on this. I believe the customers can place as many orders as they want and it should not be an issue.  In that case, we might need a Unique ID for each order instead of using the phone numbers to find the orders.

Reply:
Got it, will update `isSameOrder` and the relevant test cases